projects/igniteui-angular/src/lib/grids/selection/selection.service.ts

```diff
@@ -463,11 +462,44 @@ export class IgxGridSelectionService {
         }
         const newSelection = this.getSelectedRows().filter(r => r !== rowID);
         if (this.rowSelection.size && this.rowSelection.has(rowID)) {
-            this.selectedRowsChange.next();
```

Comment:
Removed multiple instances of sending new values to the subject _before_ calling `emitRowSelectionEvent`. The latter will eventually reach and call `selectedRowsChange.next()` after the _cancel_ check..